Utils/Dataflow/pyDKB/common/hdfs.py

```diff
@@ -207,10 +207,14 @@ def dirname(p):
     return path.dirname(p).strip()
 
 
-def join(p, filename):
-    """ Join path and filename. """
+def join(p, *args):
+    """ Join given paths. """
+    paths = []
     if p is None:
         p = ''
-    if filename is None:
-        filename = ''
-    return path.join(p, filename).strip()
+    for a in args:
+        if a is None:
+            paths.append('')
+        else:
+            paths.append(a)
+    return path.join(p, *paths).strip()
```

Comment:
And now I noticed: is there any need to leave `p` as a separate parameter? Looks like
```
join(*args):
  ...
  return path.join(*paths).strip()
```
will also work.

Reply:
True. I've left the separate parameter to make the function similar to [os.path.join()](https://docs.python.org/2/library/os.path.html#os.path.join)'s definition, but it seems that it isn't required and we can simplify our code.

Implemented the change.

Reply:
Er, I didn\`t notice this... In fact, it changes things: in `os.path.join()` (and in `posixpath.join()`, as well) it is not only about definition, it is about a required argument. So I wasn\`t exactly right: the version I have suggested is not really an equivalent to what it was (and it would be OK to point out this mistake ;) ).
So now we have a couple of options:
* rely on `posixpath.join()` exception;
* return empty string if no arguments passed;
* make the first argument required as it was (just handle it together with the rest of the arguments, like `for p in ([path] + args): ...`).

First variant is not very good, as on paper our function doesn\`t require arguments to be specified yet fails if there\`s none in deed. Which one of the rest of list is better I don\`t know: to follow the convenient of such functions or to return what user asks for (want to join nothing? Alright, get nothing).

Reply:
Well, I picked this option:

> rely on posixpath.join() exception;

but you are right - if the function fails without arguments, this fact should be reflected somewhere, for example, in its definition or description. Done this instead:

> make the first argument required as it was (just handle it together with the rest of the arguments, like for p in ([path] + args): ...).